app/src/main/java/com/eveningoutpost/dexdrip/utils/LibreTrendPoint.java

```diff
@@ -0,0 +1,49 @@
+package com.eveningoutpost.dexdrip.utils;
+
+import com.eveningoutpost.dexdrip.Models.GlucoseData;
+import com.eveningoutpost.dexdrip.Models.UserError.Log;
+
+//This class represents a per minute data from the libre.
+public class LibreTrendPoint {
+    long sensorTime; // The number of minutes from sensor start.
+    public long rawSensorValue; // The raw value of the sensor
+    int flags; // The flags that was received from the sensor (if any).
+    public GlucoseData.DataSource source; // Did we read this data from
+                                          // bluetooth or NFC? This affects the
+                                          // way that flags are interputed.
+
+    static final String TAG = "LibreTrendPoint";
+
+
+    public LibreTrendPoint(){}
+
+    // For testing
+    public LibreTrendPoint(long sensorTime, long rawSensorValue, int flags, GlucoseData.DataSource source) {
+        this.sensorTime = sensorTime;
+        this.rawSensorValue = rawSensorValue;
+        this.flags = flags;
+        this.source = source;
+    }
+
+    public String toString(){
+        return "{ sensorTime = " + sensorTime + " rawSensorValue = " + rawSensorValue +  " flags = " +   flags +  " source = " +   source +  "}";
+    }
+
+    public long getSensorTime() { return sensorTime;}
+
+    public boolean isError() {
+        boolean ret = rawSensorValue == 0;
+        if (source == GlucoseData.DataSource.FARM) {
+            ret |= flags != 800;
+        } else if (source == GlucoseData.DataSource.BLE) {
+            // Condition already checked (rawSensorValue == 0)
```

Comment:
This test looks a bit strange to me.

Reply:
In the case that data was recieved from ble, ret should be based on " rawSensorValue == 0" but it is already initilazied to this value.